type.go

```diff
@@ -1453,13 +1453,26 @@ func (t *listType) Decode(dst, _ []byte, _ encoding.Encoding) ([]byte, error) {
 // Map constructs a node of MAP logical type.
 //
 // https://github.com/apache/parquet-format/blob/master/LogicalTypes.md#maps
-func Map(key, value Node) Node {
-	return mapNode{Group{
+func Map(key, value Node, tag ...string) Node {
```

Comment:
This model looks a bit odd since it's mixing the struct tags and the manual construction of nodes.

How about keeping this logic in `nodeOf` ?

Reply:
yeah I know something was weird. updated. thanks